src/controller/eme-controller.ts

```diff
@@ -535,147 +535,156 @@ class EMEController extends Logger implements ComponentAPI {
       return;
     }
 
-    let keyId: Uint8Array | null | undefined;
-    let keySystemDomain: KeySystems | undefined;
-
-    if (
-      initDataType === 'sinf' &&
-      this.getLicenseServerUrl(KeySystems.FAIRPLAY)
-    ) {
-      // Match sinf keyId to playlist skd://keyId=
-      const json = bin2str(new Uint8Array(initData));
-      try {
-        const sinf = base64Decode(JSON.parse(json).sinf);
-        const tenc = parseSinf(sinf);
-        if (!tenc) {
-          throw new Error(
-            `'schm' box missing or not cbcs/cenc with schi > tenc`,
-          );
-        }
-        keyId = tenc.subarray(8, 24);
-        keySystemDomain = KeySystems.FAIRPLAY;
-      } catch (error) {
-        this.warn(`${logMessage} Failed to parse sinf: ${error}`);
-        return;
-      }
-    } else if (this.getLicenseServerUrl(KeySystems.WIDEVINE)) {
-      // Support Widevine clear-lead key-session creation (otherwise depend on playlist keys)
-      const psshResults = parseMultiPssh(initData);
-
-      // TODO: If using keySystemAccessPromises we might want to wait until one is resolved
+    if (!this.keyFormatPromise) {
       let keySystems = Object.keys(
         this.keySystemAccessPromises,
       ) as KeySystems[];
       if (!keySystems.length) {
         keySystems = getKeySystemsForConfig(this.config);
       }
+      const keyFormats = keySystems
+        .map(keySystemToKeySystemFormat)
+        .filter((k) => !!k) as KeySystemFormats[];
+      this.keyFormatPromise = this.getKeyFormatPromise(keyFormats);
+    }
 
-      const psshInfo = psshResults.filter((pssh): pssh is PsshData => {
-        const keySystem = pssh.systemId
-          ? keySystemIdToKeySystemDomain(pssh.systemId)
-          : null;
-        return keySystem ? keySystems.indexOf(keySystem) > -1 : false;
-      })[0];
+    this.keyFormatPromise.then((keySystemFormat) => {
+      const keySystem = keySystemFormatToKeySystemDomain(keySystemFormat);
+
+      let keyId: Uint8Array | null | undefined;
+      let keySystemDomain: KeySystems | undefined;
+
+      if (initDataType === 'sinf' && keySystem === KeySystems.FAIRPLAY) {
```

Comment:
It might be better to log when the "encrypted" event should be ignored:

```suggestion
      if (initDataType === 'sinf') {
        if (keySystem !== KeySystems.FAIRPLAY) {
          this.log(`Ignoring "${event.type}" event with init data type: "${initDataType}" for selected key-system ${keySystem}`);
        }
```

Reply:
If this occurs (aka a mismatch between getting a `sinf` type and not having a media session that is fairplay), this (and the related one), it sound like warnings to me - aka we only expect to see `sinf` if/when we're using fairplay. Maybe switch to a warn and add the word "unexpected" ("Ignoring unexpected...")